sensor/kubernetes/listener/resources/dispatcher.go

```diff
@@ -65,16 +65,17 @@ func NewDispatcherRegistry(
 	namespaces *orchestratornamespaces.OrchestratorNamespaces,
 	credentialsManager awscredentials.RegistryCredentialsManager,
 	traceWriter io.Writer,
+	storeProvider *InMemoryStoreProvider,
 ) DispatcherRegistry {
-	serviceStore := newServiceStore()
+	serviceStore := storeProvider.serviceStore
+	rbacUpdater := storeProvider.rbacStore
```

Comment:
Why not using the `storeProvider.Services()` and `RBAC()` respectively?

Reply:
Because we have a public and a private version API for the stores. The listeners need access to the private methods (E.g.: upserting services). If we pass the public interface (as in defined `common/store/types.go`) then listeners won't be able to operate the stores properly. 

Other components would only use the public interface (i.e. methods that don't need to expose internal wrappers).

I wanted to introduce a unified interface that can be used both internally and externally. But this requires a _much_ bigger change, because we would have to extract all wrappers and parsing logic from the listeners component.

Reply:
oh, I see. But if that is the case then the benefits from having the storeProvider in the current shape are much smaller than anticipated. It means, that we cannot have another storeProvider (e.g. a mock) unless we implement exactly the same set methods for the particular stores (both public and private).

Having said this is not great, I would be for adding the interfaces in one of the next PRs.